### PR TITLE
DOC: Remove outdated versionadded/changed directives

### PR DESCRIPTION
**Pull request by SMAntony**
I have gone through the files and removed all the outdated versionadded/changed directives for versions <=1.19 as instructed by @seberg in #27239 . I have run `spin docs` after making the changes to verify it the docs are being compiled properly.

### CODE REVIEW

File: doc/source/reference/arrays.classes.rst [hunk 1]
```diff
@@ -284,12 +280,6 @@ NumPy provides several hooks that classes can customize:
    NumPy may also call this function without a context from non-ufuncs to
    allow preserving subclass information.
 
-   .. versionchanged:: 2.0
-      ``return_scalar`` is now passed as either ``False`` (usually) or ``True``
-      indicating that NumPy would return a scalar.
-      Subclasses may ignore the value, or return ``array[()]`` to behave more
-      like NumPy.
-
```
Comment (charris) [inline doc/source/reference/arrays.classes.rst:292]: Why removing this?

File: numpy/lib/_twodim_base_impl.py [hunk 2]
```diff
@@ -1092,7 +1081,6 @@ def triu_indices(n, k=0, m=None):
 
     Notes
     -----
```
Comment (charris) [inline numpy/lib/_twodim_base_impl.py:1083]: There aren't any notes left.

File: numpy/ma/extras.py [hunk 3]
```diff
@@ -2225,7 +2216,6 @@ def clump_unmasked(a):
 
     Notes
     -----
```
Comment (charris) [inline numpy/ma/extras.py:2218]: No notes are left.

File: numpy/ma/extras.py [hunk 4]
```diff
@@ -2265,7 +2255,6 @@ def clump_masked(a):
...
```
Comment (charris) [inline numpy/ma/extras.py:2257]: No notes are left.

File: numpy/polynomial/chebyshev.py [hunk 5]
```diff
@@ -1222,8 +1214,6 @@ def chebval2d(x, y, c):
     Notes
     -----
```
Comment (charris) [inline numpy/polynomial/chebyshev.py:1215]: No notes are left.

File: numpy/polynomial/chebyshev.py [hunk 6]
```diff
@@ -1275,8 +1265,6 @@ def chebgrid2d(x, y, c):
...
```
Comment (charris) [inline numpy/polynomial/chebyshev.py:1266]: No notes left.

File: doc/source/reference/arrays.classes.rst (hunk 1 above)
Comment (charris) [reply doc/source/reference/arrays.classes.rst:292]: The version changed is 2.0

**Comment by charris** [comment]
Notes is an optional section, we should remove it if there are no notes. Could be another PR, but that would work best if the locations of empty notes could be easily searched for.

**Comment by SMAntony** [comment]
> Notes is an optional section, we should remove it if there are no notes. Could be another PR, but that would work best if the locations of empty notes could be easily searched for.

Ohh, I did not know that. I checked for some other places with "Notes" and if I am not wrong, some of them had empty Notes section. That's why I kept it as it is. For searching empty notes, we can use grep + regex, right? If so, I will raise a issue and fix the same with a PR.

File: doc/source/reference/arrays.classes.rst (hunk 1 above)
Comment (SMAntony) [reply doc/source/reference/arrays.classes.rst:292]: My bad, will check and revert this change.

**Comment by charris** [comment]
Thanks @SMAntony .

**Pull request merged**